flake.nix

```diff
@@ -0,0 +1,66 @@
+{
+  description = "heads flake, mostly for devshell for now";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.systems.follows = "systems";
+    };
+    systems.url = "github:nix-systems/default";
+  };
+
+  outputs = {
+    self,
+    flake-utils,
+    nixpkgs,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+    in {
+      devShell = pkgs.mkShellNoCC {
+        buildInputs = with pkgs;
+          [
+            autoconf
+            automake
+            bc
+            bison # Generate flashmap descriptor parser
+            cacert
+            ccache
+            cmake
+            cpio
+            curl
+            dtc
+            e2fsprogs
+            elfutils
+            flex
+            git
+            gnat11
+            gnumake
+            imagemagick
+            innoextract
+            libtool
+            m4
+            ncurses5 # make menuconfig
+            parted
+            perl
+            pkgconfig
+            python3
+            rsync
+            sharutils
+            texinfo
+            unzip
+            wget
+            which
+            zip
+            zlib.dev
+          ]
+          ++ [
+            # debugging/fixing/testing
+            qemu
+            vim
+          ];
```

Comment:
Looked to me like it's just concatenating two arrays - the dependencies are a single array, but for a bit of organization the true build dependencies were separated from some development tools.

Reply:
I intended to have different lists for different kinds of dependencies. Common ones needed by all boards, deps for family of boards and then these for human debug/dev/testing. This could be useful if we want to build boards as a nix flake output (`nix build .#t430-hotp-maximized` for example). It would only really be helpful if we avoid the giant-docker-image with all deps for all boards though.